src/imageio/format/avif.c

```diff
@@ -197,20 +218,36 @@ int write_image(struct dt_imageio_module_data_t *data,
   const size_t width = d->global.width;
   const size_t height = d->global.height;
   const size_t bit_depth = d->bit_depth > 0 ? d->bit_depth : 0;
+  enum avif_color_mode_e color_mode = d->color_mode;
 
-  switch (d->compression_type) {
-  case AVIF_COMP_LOSSLESS:
-    format = AVIF_PIXEL_FORMAT_YUV444;
-    break;
-  case AVIF_COMP_LOSSY:
-    format = AVIF_PIXEL_FORMAT_YUV420;
-    if (d->quality > 80) {
-        format = AVIF_PIXEL_FORMAT_YUV422;
-    }
-    if (d->quality > 90) {
-        format = AVIF_PIXEL_FORMAT_YUV444;
-    }
-    break;
+  switch (color_mode)
+  {
+    case AVIF_COLOR_MODE_RGB:
+      switch (d->compression_type)
+      {
+        case AVIF_COMP_LOSSLESS:
+          format = AVIF_PIXEL_FORMAT_YUV444;
+          break;
+        case AVIF_COMP_LOSSY:
```

Comment:
I'm sorry but you have now > 90 ok, but you need to switch the values to be equivalent to code above. That is, when > 90 you need format = AVIF_PIXEL_FORMAT_YUV444, etc...  That was the difficulty to read the code above that I wanted to avoid, as you see the code above was confusing enough :)